src/controllers/features/limited/sectorYouWorkInController.ts

```diff
@@ -47,28 +45,18 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
     const locales = getLocalesService();
     const currentUrl = BASE_URL + LIMITED_SECTOR_YOU_WORK_IN;
     try {
-        const errorList = validationResult(req);
         const session: Session = req.session as any as Session;
         const acspData : AcspData = session?.getExtraData(USER_DATA)!;
-        if (!errorList.isEmpty()) {
-            const pageProperties = getPageProperties(formatValidationError(errorList.array(), lang));
-
-            res.status(400).render(config.SECTOR_YOU_WORK_IN, {
-                previousPage: addLangToUrl(getPreviousPage(acspData), lang),
-                ...getLocaleInfo(locales, lang),
-                currentUrl,
-                ...pageProperties
-            });
+        if (req.body.sectorYouWorkIn === "OTHER") {
+            res.redirect(addLangToUrl(BASE_URL + LIMITED_WHICH_SECTOR_OTHER, lang));
         } else {
-            if (req.body.sectorYouWorkIn === "OTHER") {
-                res.redirect(addLangToUrl(BASE_URL + LIMITED_WHICH_SECTOR_OTHER, lang));
-            } else {
-                //  save data to mongodb
+            if (acspData) {
                 acspData.workSector = req.body.sectorYouWorkIn;
-                const acspDataService = new AcspDataService();
-                await acspDataService.saveAcspData(session, acspData);
-                res.redirect(addLangToUrl(BASE_URL + LIMITED_SELECT_AML_SUPERVISOR, lang));
             }
+            // save data to mongodb
+            const acspDataService = new AcspDataService();
+            await acspDataService.saveAcspData(session, acspData);
```

Comment:
If acspData is null then do we want to call saveAcspData? If not then may be this can also come in the if of line 53.